### PR TITLE
feat(bal-pagination): sticky position

### PR DESCRIPTION
**Pull request by madaluca**
added the sticky option for the pagination and updated the bal-slider

Closes #

{{short description}}

## Acceptance Criteria

- [x] Design Review by @Gagne87, @clastzoo
- [ ] Technical Review by @hirsch88, @mladenplaninicic, @nobilo
- [x] Visual test done by @hirsch88, @nobilo, @mladenplaninicic, @Gagne87, @clastzoo
- [x] Functional test done by @hirsch88, @nobilo, @mladenplaninicic
- Browser Testing by @hirsch88, @nobilo, @mladenplaninicic, @Gagne87, @clastzoo
    - Desktop
        - [x] Chrome
        - [ ] Edge
        - [x] Safari
        - [ ] Firefox
    - Tablet
        - [ ] iPad (Landscape / Portrait)
    - Mobile
        - [ ] Safari iOS
        - [ ] Chrome Preview

### CODE REVIEW

File: packages/components/src/components/bal-pagination/bal-pagination.sass [hunk 1]
```diff
@@ -2,6 +2,11 @@
   +unselectable
   display: block
 
+  +modifier(is-sticky)
+    position: sticky
+    top: 0
+    z-index: 1
```
Comment (hirsch88) [inline packages/components/src/components/bal-pagination/bal-pagination.sass:7]: - [x] did you test that with the navigation how it looks like ?

File: packages/components/src/components/bal-pagination/bal-pagination.tsx [hunk 2]
```diff
@@ -179,11 +187,13 @@ export class Pagination {
     const buttonColor = isSmall ? 'link' : 'text'
     const buttonSize = isSmall ? 'small' : ''
     const flat = isSmall
+    const isSticky = this.sticky ? 'is-sticky' : ''
 
     return (
       <Host
         class={{
           ...block.class(),
+          ...block.modifier(isSticky).class(),
```
Comment (hirsch88) [inline packages/components/src/components/bal-pagination/bal-pagination.tsx]: - [x] write `...block.modifier('is-sticky').class(this.sticky),`

**Review by hirsch88 — CHANGES_REQUESTED** [review]

File: packages/components/src/components/bal-pagination/bal-pagination.sass (hunk 1 above)
Comment (madaluca) [reply packages/components/src/components/bal-pagination/bal-pagination.sass:7]: it looks very bad :-< 
Screenshot 2022-09-21 at 21 16 59
 (added z-index: 30 on pagination element)

A solution would be to hide also the main navigation on scroll

Comment (madaluca) [reply packages/components/src/components/bal-pagination/bal-pagination.sass:7]: added top value dynamically

File: packages/components/src/components/bal-pagination/bal-pagination.tsx [hunk 3]
```diff
@@ -38,8 +39,21 @@ export class Pagination {
...
 
+  /**
+   * If 'true, the pagination will be sticky to the top
+   */
+  @Prop() sticky = false
+
+  /**
+   * If sticky, the top position will be determined by this value
+   */
+  @Prop() top = 0
```
Comment (hirsch88) [inline packages/components/src/components/bal-pagination/bal-pagination.tsx:55]: - [x] add a watcher to this prop. so when it changes we update the this.el.style.top as well

**Review by hirsch88 — CHANGES_REQUESTED** [review]

File: packages/components/src/components/bal-pagination/bal-pagination.tsx (hunk 3 above)
Comment (madaluca) [reply packages/components/src/components/bal-pagination/bal-pagination.tsx:55]: done

File: packages/components/src/components/bal-pagination/bal-pagination.tsx [hunk 4]
```diff
@@ -48,6 +67,9 @@ export class Pagination {
 
   componentWillLoad() {
     this._value = this.value
+    if (this.sticky) {
```
Comment (hirsch88) [inline packages/components/src/components/bal-pagination/bal-pagination.tsx]: - [x] add the if to the watcher

**Review by hirsch88 — CHANGES_REQUESTED** [review]

**Review by hirsch88 — APPROVED** [review]

**Pull request merged**